hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneServiceProvider.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.s3;
+
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.ozone.OmUtils;
+import org.apache.hadoop.security.SecurityUtil;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * This class creates the OM service .
+ */
+@ApplicationScoped
+public class OzoneServiceProvider {
+
+  private static final AtomicReference<Text> OM_SERVICE_ADD =
+      new AtomicReference<>();
+
+  @Inject
+  private OzoneConfiguration conf;
+
+
+  @Produces
+  public Text getService() {
+    if (OM_SERVICE_ADD.get() == null) {
+      OM_SERVICE_ADD.compareAndSet(null,
+          SecurityUtil.buildTokenService(OmUtils.getOmAddressForClients(conf)));
```

Comment:
Now we have HA, and we are using getOmAddressForClients, this takes om address from ozone.om.address. So, do we need to find leader OM and set it. Don't have complete context on the security, and how this is used. So, just want to know how this works?

Reply:
We are adding om serviceso that TokenSelctor can select right token for OM. See OzoneDelegationTokenSelector for more details. For OM HA we might have to update this. HDDS-1230 to handle that.